tests/unittests/test_cache.py

```diff
@@ -359,3 +360,474 @@ def f2(self, version):
 
     assert b1.f1.name.endswith("f1")
     assert b1.f2.name.endswith("f2")
+
+
+def test_binary_formatter(tmp_path):
+    """Test BinaryFormatter with binary data."""
+    from sarc.cache import BinaryFormatter
```

Comment:
I'm not against inside function imports but I'm not sure if it is needed here?

Reply:
No it's not.  I though that I removed all in-function imports, but it seems I forgot a lot so I removed them